src/EditorFeatures/Core/CommandHandlers/InteractivePasteCommandHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Windows;
+using Microsoft.VisualStudio.Text.Operations;
+using Microsoft.VisualStudio.Utilities;         
+using Microsoft.CodeAnalysis.Editor.Commands;
+using Microsoft.CodeAnalysis.Editor.Interactive;
+
+namespace Microsoft.CodeAnalysis.Editor.CommandHandlers
+{
+    [ExportCommandHandler("InteractivePasteCommandHandler", ContentTypeNames.RoslynContentType)]
+    [Order(After = PredefinedCommandHandlerNames.FormatDocument)]
+    internal sealed class InteractivePasteCommandHandler : ICommandHandler<PasteCommandArgs>
+    {
+        private readonly IEditorOperationsFactoryService _editorOperationsFactoryService;
+
+        [ImportingConstructor]
+        public InteractivePasteCommandHandler(IEditorOperationsFactoryService editorOperationsFactoryService)
+        {
+            _editorOperationsFactoryService = editorOperationsFactoryService;
+        }
+
+        public void ExecuteCommand(PasteCommandArgs args, Action nextHandler)
+        {
+            if (args.TextView.TextBuffer.ContentType.TypeName != PredefinedInteractiveContentTypes.InteractiveContentTypeName &&
```

Comment:
Upon further consideration, it might be nice to leave a comment to the effect of:  "InteractiveWindow handles pasting by itself, which including checks for buffer types, etc."